### PR TITLE
Convert Gradle to Maven Build

### PR DESCRIPTION
**Pull request by ALRubinger**
Ready to go for review. Do not merge until the web5-kt PR below is merged and a proper release is cut there, and updated here. 

I'll avoid repetition here. Reasoning and layout is same as extensively detailed in:

* https://github.com/TBD54566975/web5-kt/pull/261

Deployed a Release Candidate build: https://blockxyz.jfrog.io/artifactory/tbd-oss-releases-maven2/xyz/block/tbdex/1.0.0-maven-rc-1/

This has been validated as working in `developer.tbd.website`:

* https://github.com/TBD54566975/developer.tbd.website/tree/alrubinger/tbdex-upgrade-v1-kt-maven

Accomplishes the principal goal: make it so that consumers of `tbdex-kt` get versioning aligned with `web5-kt` without the need to do any manual dependency coordination by appdevs. Proven by the dev site testsuite POM: https://github.com/TBD54566975/developer.tbd.website/blob/alrubinger/tbdex-upgrade-v1-kt-maven/site/testsuites/testsuite-kotlin/pom.xml#L30

### CODE REVIEW

File: .github/workflows/ci.yaml
Comment (jiyoontbd) [inline .github/workflows/ci.yaml:1]: can you change this filename to `ci.yml` please so i can see the diff 🙏

Comment (jiyoontbd) [reply .github/workflows/ci.yaml:1]: also for consistency

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+    <dependencies>
+
+      <!-- Import Dependencies -->
+      <!-- Web5 BOM; sets appropriate deps for the platform-->
+      <dependency>
+        <groupId>xyz.block</groupId>
+        <artifactId>web5-parent</artifactId>
+        <version>${version.xyz.block.web5}</version>
+        <type>pom</type>
+        <scope>import</scope>
```
Comment (jiyoontbd) [inline pom.xml:137]: is this the thing that does inheritance?

File: config/dokka-configuration.json
Comment (jiyoontbd) [inline config/dokka-configuration.json:1]: same comment as https://github.com/TBD54566975/web5-kt/pull/261/files#r1552236144

**Review by jiyoontbd — COMMENTED** [review]
looks mostly consistent with your web5-kt pr, so same comments should apply here too!

**Comment by ALRubinger** [comment]
Must document in `README`: Updates to new `web5-kt` must be done in two places in the parent `pom.xml`: when defining the `<parent>`, and when defining `<version.xyz.block.tbdex>`.

**Comment by ALRubinger** [comment]
Add to `README`: note the `distribution` package.

File: README.md [hunk 2]
```diff
@@ -1,259 +1,286 @@
...
+[![License](https://img.shields.io/github/license/TBD54566975/tbdex-kt)](https://github.com/TBD54566975/tbdex-kt/blob/main/LICENSE) [![CI](https://github.com/TBD54566975/tbdex-kt/actions/workflows/ci.yaml/badge.svg)](https://github.com/TBD54566975/tbdex-kt/actions/workflows/ci.yaml) [![Coverage](https://img.shields.io/codecov/c/gh/tbd54566975/tbdex-kt/main?logo=codecov&logoColor=FFFFFF&style=flat-square&token=YI87CKF1LI)](https://codecov.io/github/TBD54566975/tbdex-kt) [![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/TBD54566975/tbdex-kt/badge)](https://securityscorecards.dev/viewer/?uri=github.com/TBD54566975/tbdex-kt)
 
-This repo contains 2 jvm packages:
+This repo contains packages:
 
 * [`/protocol`](./protocol/) - create, parse, verify, and validate the tbdex messages and resources defined in
   the [protocol draft specification](https://github.com/TBD54566975/tbdex/blob/main/specs/protocol/README.md)
 * [`/httpclient`](./httpclient) - An HTTP client that can be used to send tbdex messages to PFIs
+* [`/httpserver`](./httpserver) - Base implementation of a tbDEX HTTP server responsible for handling RFQs, orders, and other interactions
+
```
[the remark this answers is not in the document]
Comment (ALRubinger) [reply README.md:11]: Noted in one of my comments on this PR that we need to handle, yes - thx

[the remark this answers is not in the document]
Comment (ALRubinger) [reply README.md:11]: Done.

**Review by jiyoontbd — APPROVED** [review]
🥳 🌟 🎉 🥇

**Review by leordev — APPROVED** [review]

**Pull request merged**